sources/settings-migrations/v1.21.0/add-hostname-override-source/src/main.rs

```diff
@@ -0,0 +1,21 @@
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added one new settings for configuring the override method for the method used to determine
+/// the node
+fn run() -> Result<()> {
+    migrate(AddSettingsMigration(&[
+        "settings.kubernetes.hostname-override-source",
+    ]))
+}
```

Comment:
Reading this, my concern is that current hosts (with no `hostname-override-source` set) will have their default generator behavior changed on update because we don't populate the new default value here if it's empty.

But I suppose that's not an issue because the `hostname-override` will have been set on first-boot and therefore the `hostname-override` generator won't run again and this value is meaningless.

Reply:
Yeah, `hostname-override` is a weird property with the way it auto-sets.  For variants where that occurs, changing `hostname-override-source` after the instance has no effect.

Reply:
The core kit migration and moving `pluto` away from a setting generator has made this hard to follow.

In 1.20.x, `aws-k8s-1.25+` had a [`hostname-override` setting generator](https://github.com/bottlerocket-os/bottlerocket/blob/a3e8bda116dcf4b06fdf6e4cea42da1a005e926f/sources/models/shared-defaults/kubernetes-aws-external-cloud-provider.toml#L8) defined that would call `pluto private-dns-name` to generate a hostname override.

In this PR, `aws-k8s-1.25+` will instead get `hostname-override-source = "private-dns-name"` (via `storewolf` and the newer defaults), which triggers the equivalent path in the newer `pluto`. But newer `pluto` will also skip over all of this if `hostname-override` is already set, which it would be on upgrade from an existing `aws-k8s-1.25+` host.